packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -58,6 +59,8 @@ class BuildIOSCommand extends BuildSubCommand {
 
   @override
   Future<FlutterCommandResult> runCommand() async {
+    Cache.releaseLockEarly();
```

Comment:
The cache is only for downloading artifacts (I think). Could we release the cache automatically after checking for cache updates? that way it wouldn't need to be done adhoc in each subcommand, and would be able to skip the cache for commands that don't download anything (like --version)

Reply:
That makes sense.  The one-off I see is test command explicitly running [`await globals.cache.updateAll(await requiredArtifacts);`](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/commands/test.dart#L159) which is causing it to check for updates twice.

(Introduced at https://github.com/flutter/flutter/pull/30254/files#diff-57bf7f0d5a70ab392ae3047369120c83R102).

Reply:
Also `precache` probably needs it re-locked.